haas/cli.py

```diff
@@ -213,49 +262,110 @@ def user_remove_project(user, project):
 @cmd
 def project_create(project):
     """Create a <project>"""
-    url = object_url('project', project)
-    do_put(url)
+    try:
+        ret = h.project_create(project)
+        print "    SUCCESS: Project %s created " % project
+    except DuplicateName as e:
+        print e.message
+    except UnknownError as e:
+        print e.message
 
 @cmd
 def project_delete(project):
     """Delete <project>"""
-    url = object_url('project', project)
-    do_delete(url)
+    try:
+        ret = h.project_delete(project)
+        print "Success: project %s deleted. " % project
+    except IncompleteDependency as e:
+        print e.message
+    except UnknownError as e:
+        print e.message
+
+#    url = object_url('project', project)
+#    do_delete(url)
 
 @cmd
 def headnode_create(headnode, project, base_img):
     """Create a <headnode> in a <project> with <base_img>"""
-    url = object_url('headnode', headnode)
-    do_put(url, data={'project': project,
-                      'base_img': base_img})
+
+    try:
+        ret = h.headnode_create(headnode, project, base_img)
+        print "Success: Headnode Created for project  %s " % project
+    except DuplicateName as e:
+        print e.message
+    except IncompleteDependency as e:
+        print e.message
+    except UnknownError as e:
+        print e.message
+
 
 @cmd
 def headnode_delete(headnode):
     """Delete <headnode>"""
-    url = object_url('headnode', headnode)
-    do_delete(url)
+
+    try:
+        ret = h.headnode_delete(headnode)
+        print "Success: Headnode %s deleted." % project
+    except IncompleteDependency as e:
+        print e.message
+    except UnknownError as e:
+        print e.message
 
 @cmd
 def project_connect_node(project, node):
     """Connect <node> to <project>"""
-    url = object_url('project', project, 'connect_node')
-    do_post(url, data={'node': node})
+
+    try:
+        ret = h.project_connect_node(project, node)
+        print "Success: Node %s connected to project %s " % (node, project)
+    except DuplicateName as e:
```

Comment:
Instead of duplicating these exceptions in every function, why not surround the overall CLI dispatch function with them?